rfcs/20251105-rms-norm-graph-op/README.md

```diff
@@ -0,0 +1,321 @@
+2. No beta/bias parameter (distinguishes from LayerNorm).
+3. The `rrms` output is optional and typically used when backward pass is needed.
+   It stores the reciprocal of RMS for each sample, enabling efficient gradient
+   computation without recomputing RMS.
+
+**Example Usage:**
+
+```cpp
+// Create RMSNorm op for transformer layer
+// Input: [batch=32, seq_len=128, hidden_dim=768]
+// Normalize over last dimension
+auto rms_norm_op = graph::op_t(graph::op::kind::RMSNorm);
+rms_norm_op.set_attr<int64_t>(graph::op_attr::axis, -1);
+rms_norm_op.set_attr<float>(graph::op_attr::epsilon, 1e-5f);
+```
+
+**Pros:**
+
+1. Clear semantics with one-to-one mapping to framework operations.
+2. Simpler parameter set than LayerNorm (no mean, no beta).
+3. Distinguishes RMSNorm from LayerNorm at the API level, making fusion
+   patterns explicit.
+4. Easier for frameworks to integrate.
+
+**Cons:**
+
+1. Adds another normalization op to maintain.
+2. Requires separate fusion patterns from LayerNorm.
+
+### Option 2: Extend LayerNorm with a flag to indicate RMSNorm behavior
+
+| LayerNorm          | oneDNN Graph API   |
+| ------------------ | ------------------ |
+| input              | src                |
+| input              | gamma (optional)   |
+| input              | beta (optional)    |
+| attribute          | keep_stats         |
+| attribute          | begin_norm_axis    |
+| attribute          | use_affine         |
+| attribute          | epsilon            |
+| attribute          | rms_norm           |
+| output             | dst                |
+| output             | mean (optional)    |
+| output             | variance (optional)|
+
+Add a new attribute `rms_norm` (default=false) to LayerNorm. When `rms_norm=true`,
+the operation performs RMS normalization (no mean subtraction, no variance computation).
+The `beta` parameter is ignored and `mean`/`variance` outputs are not generated when
+`rms_norm=true`.
+
+**Pros:**
+
+1. Reuses existing LayerNorm infrastructure.
+2. Fewer operations to maintain.
+3. Single operation covers both LayerNorm and RMSNorm use cases.
+
+**Cons:**
+
+1. Overloaded semantics - LayerNorm becomes ambiguous.
+2. Framework integration is less intuitive (need to set flags).
+
+## Decision
+
+**Choose Option 1** - Add RMSNorm as a dedicated operation.
+
+**Reasoning:**
+
+1. **Clear Semantics**: RMSNorm has fundamentally different semantics from
+   LayerNorm (no mean subtraction, no bias). A dedicated op makes this explicit.
+
+2. **Framework Adoption**: All major LLM frameworks treat RMSNorm as a distinct
+   operation. PyTorch, OpenVINO, and ONNX all have dedicated RMSNorm
+   implementations.
+
+5. **Growing Usage**: RMSNorm is becoming the standard for new LLMs (LLaMA
+   family, Mistral, Gemma). This trend justifies a first-class operation.
+
+6. **Simpler API**: Fewer attributes and inputs than trying to overload
+   LayerNorm. No optional mean/variance outputs needed.
+
+## Proposal for RMSNorm Backward
```

Comment:
Maybe fuse both propagation kinds into a single proposal? I don't see why they would diverge in implementation.